tests/java_integers.rs

```diff
@@ -0,0 +1,81 @@
+extern crate error_chain;
+extern crate jni;
+
+use error_chain::ChainedError;
+use jni::{InitArgsBuilder, JavaVM, JNIEnv};
+use jni::objects::JObject;
+use jni::objects::JValue;
+use jni::sys::JNI_VERSION_1_8;
+
+fn print_exception(env: &JNIEnv) {
+    let exception_occurred = env.exception_check().unwrap_or_else(
+        |e| panic!(format!("{:?}", e)),
+    );
+    if exception_occurred {
+        env.exception_describe().unwrap_or_else(
+            |e| panic!(format!("{:?}", e)),
+        );
+    }
+}
+
+#[test]
+fn test_java_integers() {
+    unsafe {
+        let jvm_args = InitArgsBuilder::new()
+            .version(JNI_VERSION_1_8)
+            .option("-Xcheck:jni")
+            .option("-Xdebug")
+            .build()
+            .unwrap_or_else(|e| {
+                panic!(format!("{}", e.display_chain().to_string()));
+            });
+
+        let (jvm, env) = JavaVM::new(jvm_args).unwrap_or_else(|e| {
+            panic!(format!("{}", e.display_chain().to_string()));
+        });
+
+        for int_value in -10..10 {
+            for array_length in 1..50 {
+                env.push_local_frame(0).unwrap_or_else(|e| {
+                    print_exception(&env);
+                    panic!(format!("{}", e.display_chain().to_string()));
+                });
+
+                let integer_value = env.new_object("java/lang/Integer", "(I)V", &[JValue::Int(int_value)])
+                    .map(|x| JObject::from(x))
+                    .and_then(|x| env.new_global_ref(x))
```

Comment:
Why create global references if you don't need to store them or pass between threads? BTW, in this case (when you create a JVM and keep things in a single main thread) *you* are responsible to delete even the local references (because there is no such thing as a return from a native method: you start and work in a native method, therefore you must tell the GC when you are done with a reference to an object). `env.with_local_frame`, `Push/PopLocalFrame` or `AutoLocal`s help to do that.

Reply:
Thanks! I was not sure about that